sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala

```diff
@@ -470,6 +474,14 @@ class DataFrameReader private[sql](sparkSession: SparkSession) extends Logging {
     sparkSession.internalCreateDataFrame(parsed, schema, isStreaming = jsonDataset.isStreaming)
   }
 
+  private def checkJsonSchema(schema: StructType): Unit = schema.existsRecursively{
```

Comment:
Can we deduplicate this with JSON expressions? You could move the logic into ExprUtils.

Reply:
Thanks for the advice! I have refactored it.